changelog/3408.feature.rst

```diff
@@ -0,0 +1,4 @@
+The returned list of `sunpy.map.Map` objects is now sorted by filename when passing a
+directory or glob pattern to `sunpy.map.MapFactory`. Single character wildcards and
+character ranges can now also be used as glob patterns. `sunpy.map.Map` also now accepts
```

Comment:
I think the acceptance of `Path` objects is worthy of it's own changelog entry.

Reply:
The changelog now has three separate entries: 
1. sorted output from glob and directories
2. support for glob patterns other than `*`
3. support for `pathlib.Path`